configure.ac

```diff
@@ -377,61 +377,199 @@ fi
+   # It is more reliable to test linking with doupdate(), which
+   # refresh() would call internally.
+   if test "$htop_libcurses_status" -eq 0; then
+      AC_MSG_CHECKING([for doupdate in $htop_msg_linker_flags])
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+int doupdate(void);
+         ]], [[
+doupdate();
+         ]])], [
+            AC_MSG_RESULT(yes)
+         ], [
+            AC_MSG_RESULT(no)
+            htop_libcurses_status=1
+         ])
+   fi
 
+   if test "x$htop_libcurses_status$enable_unicode" = x0yes; then
+      AC_MSG_CHECKING([for mvadd_wchnstr in $htop_msg_linker_flags])
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+/* int mvadd_wchnstr(int y, int x, const cchar_t *wchstr, int n); */
+int mvadd_wchnstr(int y, int x, const void* wchstr, int n);
+      ]], [[
+mvadd_wchnstr(0, 0, (void*)0, 0);
+      ]])], [
+         AC_MSG_RESULT(yes)
+      ], [
+         AC_MSG_RESULT(no)
+         htop_libcurses_status=1
+      ])
+   fi
+
+   if test "$htop_libcurses_status" -eq 0; then
+      AM_CFLAGS="$AM_CFLAGS $htop_libcurses_cflags"
+      htop_curses_pkg_found=yes
+      if test "x$enable_unicode" = xyes; then
+         AC_DEFINE([HAVE_LIBNCURSESW], 1, [libncursesw is present])
+      else
+         # TODO: rename the preprocessor token
+         AC_DEFINE([HAVE_LIBNCURSES], 1, [libcurses is present])
+      fi
+   else
+      LIBS=$htop_save_LIBS
+   fi
+   CFLAGS=$htop_save_CFLAGS
+   return "$htop_libcurses_status"
+} # htop_check_libcurses_capability
+
+htop_curses_pkg_found=no
+
+if test "x${CURSES_CFLAGS+y}${CURSES_LIBS+y}" = xyy; then
+   curses_pkg_names=""
+   htop_check_libcurses_capability "$CURSES_CFLAGS" "$CURSES_LIBS"
+fi
+
+for curses_name in $curses_pkg_names; do
+   htop_config_libs_retrieved=no
+   m4_ifdef([PKG_PROG_PKG_CONFIG], [
+   if test "x$PKG_CONFIG" != x; then
+      echo running pkg-config for "$curses_name" >&AS_MESSAGE_LOG_FD
+      if $PKG_CONFIG --exists --print-errors "$curses_name" >&AS_MESSAGE_LOG_FD 2>&AS_MESSAGE_LOG_FD; then
+         htop_config_cflags=`$PKG_CONFIG --cflags "$curses_name" 2>/dev/null`
+         if test "$?" -eq 0; then
+            htop_config_libs=`$PKG_CONFIG --libs "$curses_name" 2>/dev/null`
+            if test "$?" -eq 0; then
+               AC_MSG_NOTICE([found $curses_name configuration through $PKG_CONFIG])
+               htop_config_libs_retrieved=yes
+	    fi
+         fi
+      fi
+   fi
+   ])
+
+   if test "$htop_config_libs_retrieved" = no; then
+      echo running "${curses_name}-config" for "$curses_name" >&AS_MESSAGE_LOG_FD
+      if "${curses_name}-config" --version >/dev/null 2>&AS_MESSAGE_LOG_FD; then
+         htop_config_cflags=`"${curses_name}-config" --cflags 2>/dev/null`
+         if test "$?" -eq 0; then
+            htop_config_libs=`"${curses_name}-config" --libs 2>/dev/null`
+            if test "$?" -eq 0; then
+               AC_MSG_NOTICE([found $curses_name configuration through ${curses_name}-config])
```

Comment:
Shouldn't this be `AC_MSG_RESULT`?

Reply:
I was experimenting the output message when configuring. The idea was to show where we found the linker flags (CFLAGS and LIBS, in particular) and to leave a copy of the message in `config.log`. If I use `AC_MSG_RESULT`, then the message in `config.log` would have the `result:` prefix. The wording would look weird as in this example:

```text
result: found ncursesw6 configuration through ncursesw6-config
```

Do you have a better wording suggestion?

Or alternatively I can just use `echo text >&AS_MESSAGE_FD` so it prints the same line in standard output and in `config.log`.

Reply:
I did `AC_MSG_RESULT("found $LIBNAME ($METHOD)")` in #1506 and I think that's completely okay. Similar to what some of the other messages do too.

Reply:
I just got the idea. The message could look like this in `config.log`...

```text
retrieving $curses_name configuration through $PKG_CONFIG
[pkg-config output]
result: found $curses_name configuration through $PKG_CONFIG
```